### PR TITLE
Add Hexadecimal state handling to excel_to_xtce

### LINKED ISSUE #1158: excel to xtce should handle hexadecimal state values
### Summary

Some telemetry definition housekeeping state values are listed in hexadecimal form. `space-packet-parser` requires states to be in decimal form. This PR adds a function to ensure that the state value is an integer or a hexadecimal string is converted to an integer.

### PR DESCRIPTION
**Pull request by sdhoyt**
# Change Summary

## Overview
Some telemetry definition housekeeping state values are listed in hexadecimal form. `space-packet-parser` requires states to be in decimal form. This PR adds a function to ensure that the state value is an integer or a hexadecimal string is converted to an integer.

Closes #1158

### CODE REVIEW

File: imap_processing/ccsds/excel_to_xtce.py [hunk 1]
```diff
@@ -391,8 +391,40 @@ def _add_state_conversion(self, row: pd.Series, parameter_type: Et.Element) -> N
...
+        dict
+            The dictionary for the state.
+        """
+        value = state["value"]
+        # convert hex string to int
+        if isinstance(value, str) and value.startswith("0x"):
+            state["value"] = int(value, 16)
+            return state
+        # return if already an int
+        elif isinstance(value, int):
```
Comment (vmartinez-cu) [inline imap_processing/ccsds/excel_to_xtce.py]: If integers are more common than hex values, you could switch the if/elif logic to check if value is an integer first. There might be a slight performance gain there.

**Review by vmartinez-cu — APPROVED** [review]
Nice job! One minor comment, but not something that would block a merge

File: imap_processing/ccsds/excel_to_xtce.py (hunk 1 above)
Comment (sdhoyt) [reply imap_processing/ccsds/excel_to_xtce.py]: Good point, I'll update that

File: imap_processing/tests/ccsds/test_excel_to_xtce.py [hunk 2]
```diff
@@ -197,7 +197,7 @@ def xtce_excel_file(tmp_path):
     states = {
         "packetName": ["TEST_PACKET"] * 2,
         "mnemonic": ["VAR_STATE"] * 2,
-        "value": [0, 1],
+        "value": [0, "0x1"],
```
Comment (tech3371) [inline imap_processing/tests/ccsds/test_excel_to_xtce.py]: can you add this hex test as another one without deleting that one? I think we need to test that also

**Review by tech3371 — COMMENTED** [review]
Otherwise, LGTM!

**Review by tmplummer — APPROVED** [review]
LGTM

**Pull request merged**